client/dashboard/sites/staging-site-sync-modal/test/index.test.tsx

```diff
@@ -0,0 +1,402 @@
+	} );
+} );
+
+describe( 'Domain Confirmation', () => {
+	test( 'shows domain confirmation field when syncing to production', () => {
+		mockUseQuery( createMockSite(), createMockStagingSite() );
+
+		renderModal( { syncType: 'push', environment: 'staging' } );
+
+		expect( screen.getByLabelText( 'Type the site domain to confirm' ) ).toBeInTheDocument();
+	} );
+
+	test( 'does not show domain confirmation when not syncing to production', () => {
+		mockUseQuery( createMockSite(), createMockStagingSite() );
+
+		renderModal( { syncType: 'push', environment: 'production' } );
+
+		expect( screen.queryByLabelText( 'Type the site domain to confirm' ) ).not.toBeInTheDocument();
+	} );
+
+	describe( 'Database Selection', () => {
+		test( 'database checkbox shows warning when checked', async () => {
+			mockUseQuery( createMockSite(), createMockStagingSite() );
+
+			const { user } = setup();
+
+			const databaseCheckbox = screen.getByLabelText( 'Database' );
+			expect( databaseCheckbox ).toBeInTheDocument();
+
+			expect(
+				screen.queryByText( /Warning! Database will be overwritten/i )
+			).not.toBeInTheDocument();
+
+			await user.click( databaseCheckbox );
+
+			await waitFor( () => {
+				expect( screen.getByText( /Warning! Database will be overwritten/i ) ).toBeInTheDocument();
+			} );
+
+			expect( screen.getByText( /overwrite the site database/i ) ).toBeInTheDocument();
+		} );
+	} );
+
+	test( 'shows WooCommerce warning when syncing WooCommerce site to production', async () => {
+		const siteWithWoo = createMockSite( { options: { woocommerce_is_active: true } } );
+		const stagingSiteWithWoo = createMockStagingSite( {
+			options: { woocommerce_is_active: true },
+		} );
+
+		mockUseQuery( siteWithWoo, stagingSiteWithWoo );
+
+		const { user } = setup( { syncType: 'push', environment: 'staging' } );
+
+		const databaseCheckbox = screen.getByLabelText( 'Database' );
+
+		await user.click( databaseCheckbox );
+
+		await waitFor( () => {
+			expect( screen.getByText( /WooCommerce installed/i ) ).toBeInTheDocument();
+		} );
+	} );
+} );
+describe( 'Form Submission', () => {
+	test( 'submit button is disabled when domain confirmation is required but not provided', () => {
+		mockUseQuery( createMockSite(), createMockStagingSite() );
+
+		renderModal( { syncType: 'push', environment: 'staging' } );
+
+		const submitButton = screen.getByRole( 'button', { name: 'Push' } );
+		expect( submitButton ).toBeDisabled();
+	} );
+
+	test( 'submit button is enabled when domain confirmation matches', async () => {
+		mockUseQuery( createMockSite(), createMockStagingSite() );
+
+		renderModal( { syncType: 'push', environment: 'staging' } );
+
+		const domainInput = screen.getByLabelText( 'Type the site domain to confirm' );
+		const user = userEvent.setup();
+
+		await user.type( domainInput, 'test-site' );
+
+		// Just check the input has the value, form validation is complex
+		expect( domainInput ).toHaveValue( 'test-site' );
+	} );
```

Comment:
Should we check the submit button is actually enabled? Should we also check the actual submission of the form?